src/Core/Module/ModuleRepository.php

```diff
@@ -150,6 +151,10 @@ public function getModule(string $moduleName): ModuleInterface
 
         $isValid = $filemtime > 0 && $this->moduleDataProvider->isModuleMainClassValid($moduleName);
         $attributes = $this->getModuleAttributes($moduleName, $isValid);
+        if (empty($attributes)) {
+            $isValid = false;
+        }
+        $attributes = array_merge(['name' => $moduleName], $attributes);
```

Comment:
```suggestion
        $attributes['name'] = $moduleName;
```

Reply:
Hi @atomiix 
Why array merge ?

Reply:
@PululuK, why not? 😄

Reply:
😁 yep why not ... But my question was really sincere, I really wanted to understand. Maybe you didn't want to overwrite the name key if it already existed ?

Reply:
Nothing more than personal preference 😄 I think the purpose is clearer here when we use `array_merge` than just an new assignment. Also, it puts `name` as the first key of the array (easier to debug).

Reply:
Okay but I think a new assignment is faster than array_merge (involves more processing), of course it's micro optimizations, but when it goes in a loop we can save a few seconds.

Reply:
I saw in [this thread](https://stackoverflow.com/questions/57192782/array-merge-vs-direct-array-injection-performance), that `array_merge` was faster 🤷‍♂️ 😅

Reply:
It's not in the same context, in your case :

- array has only one element(equivalent to add a simple variable and assigning a value to it)

Reply:
According the thread https://stackoverflow.com/questions/57192782/array-merge-vs-direct-array-injection-performance
![image](https://user-images.githubusercontent.com/16455155/166905730-06ca4d16-13cd-4a5a-a854-3c8f350c7dc9.png)

Remember that each "if" in this sentence represents a calculation

Reply:
You're right!
But as you said, this is micro optimization, I think we can keep it?